Source/WebCore/xml/XPathPredicate.cpp

```diff
@@ -281,7 +281,12 @@ Value Union::evaluate() const
 
 bool evaluatePredicate(const Expression& expression)
 {
-    Value result(expression.evaluate());
+    EvaluationContext clonedContext(Expression::evaluationContext());
+    Value result = [&] {
+        SetForScope<EvaluationContext> contextForScope(Expression::evaluationContext(), clonedContext);
+        return expression.evaluate();
+    }();
+    Expression::evaluationContext().hadTypeConversionError |= clonedContext.hadTypeConversionError;
```

Comment:
Seems like we might want a helper that propagates "all the right things". But maybe "all the right things" is only ever this single boolean.

Reply:
For now, this boolean is this only thing we have propagated.